src/map/Map.tsx

```diff
@@ -30,13 +33,14 @@ export default function({ selectedPath, paths, queryPoints, bbox, mapStyle }: Ma
                 Dispatcher.dispatch(new MapIsLoaded())
             }
         )
-        mapWrapper.fitBounds(bbox)
+        if (prevViewPort)
+            mapWrapper.setViewPort(prevViewPort)
```

Comment:
This code is what was missing for #72. When we re-create the map instance we have to restore the previous viewport. However, just doing this is not enough, because without the changes in `App.tsx` there would be another call to `fitBounds` that sets the info or route bbox again.